Changed fd_create to fd-create

AryanGodara opened the pull request:
```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```
Changed the name of veracruz/sdk/rust-examples/fd_create to veracruz/sdk/rust-examples/fd-create
This is so that the naming convention matches with all the other folders in the directory.

dominic-mulligan-arm commented:
Hi @AryanGodara, and thanks for your first Veracruz PR.  Welcome to the project!  🎉

# Developer Certificate of Origin

As this is your first commit to the Veracruz repository can you please certify that you have authorisation to contribute code using the Developer Certificate of Origin?  This is a requirement of the Veracruz organization, which we agreed with the Linux Foundation when we were adopted.  To do this, it's enough to include the following text in your PR message, at the top:

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

# Signed commits

Can I ask that you sign all of your commits using GPG?  We adopted this rule a while ago as a security/integrity mechanism and code cannot be merged into the main Veracruz repository without this signing.  If you don't know how to do this, then follow these instructions:

1. Generate a new GPG key: `gpg --gen-key` and follow the instructions asking for your name, email, etc.
2. List the GPG keys available: `gpg --list-secret-keys --keyid-format=long`.  Make a note of the ID of the key you’ve just generated (it’s a long string of capital letters and numbers),
3. Export the public key component of the generated key to stdout : `gpg --armor --export $KEYID` where `$KEYID` is the long string of letters and numbers, above.
4. Go to your Github profile’s settings page, go to the Keys section, add a new GPG key and copy-paste the public key printed out above.
5. Tell Git to use this key when signing commits: `git config --global user.signingkey $KEYID` , again where `$KEYID` is the long string of letters and numbers, above.
6. Tell Git to always sign your commits: `git config --global commit.gpgsign true` (Note, as an alternative, you can choose not to set this but then you will always need to commit using `git commit -S`)
7. To test if your commits are being signed correctly, commit something on a local copy, and then use: `git log --show-signature`.  Note that Github signs their own merge commits made to a repository when a pull request is merged, but the command above may not show the signed commits correctly (they’ll be highlighted in red) as you need to import Github’s public key to verify the signature.  To do this, import Github’s key by: `curl https://github.com/web-flow.gpg | gpg --import` Showing the signatures in Git log should now work, and Git should recognise that Github’s merge commits were signed by Github (they should now be blue).

Once you've setup GPG signing, you usually have to `export GPG_TTY=$(tty)` in your terminal (or you'll get an error about a missing `tty`) prior to commiting/rebasing.  Once that is done, edit your PR to sign all commits and push again.

dominic-mulligan-arm commented:
(Can you also edit `AUTHORS.markdown` to add your name, and add the changed file to this PR, too?)

AryanGodara commented:
@dominic-mulligan-arm I've included the text in the PR message. I just need some time to figure out the GPG part. I'll do that and also update AUTHORS.markdown too, as soon as possible.👍

dominic-mulligan-arm commented:
Excellent, thanks @AryanGodara!  Are you running Ubuntu Linux?  Installing GPG can be done with `sudo apt-get install gnupg`.

AryanGodara commented:
> 

I'm using Debian with WSL2 on windows 11. Since it'll take time to dual boot ubuntu and I may lose some important data, I plan on doing it once the contribution period is over. 
But thank you for providing the steps in such detail, it has really made my work much easier!!

AryanGodara commented:
@dominic-mulligan-arm 
All the requested changes are done now.
Sorry for taking so long, WSL2 and Debian are super super slow for git.
I will make more corrections if required!

dominic-mulligan-arm commented:
@AryanGodara unfortunately all of the commits in a PR need to be signed in order to be allowed into our `main` (if you click on the "Commits" tab at the top, you will see that two of the commits are not marked as "Verified", meaning merge will be blocked by our branch commit rules).  To fix this, you can do the following steps:

1. Do `git log` and find the hash of the commit immediately prior to your first commit.  Call this `$HASH`.
2. Do `git reset --soft $HASH` to do a "soft reset" on your branch to the state immediately prior to your first change, while keeping all of your changes in the Git staging area.
3. Do `git commit` to create a new, single commit capturing all of your changes, now signed by GPG.
4. Do `git push --force-with-lease` to update the pull request.

dominic-mulligan-arm commented:
Also, it appears that the CI is failing due to build issues in the SDK.  If you do `git grep fd_create` from within your Veracruz root repository, it should list the files that are still referring to the old `fd_create` which you have now renamed.  Fixing those (which I presume are Makefiles, or similar build scripts) and adding them to the PR should solve the CI issue.

AryanGodara commented:
@dominic-mulligan-arm 
I have fixed the first issue with the commits, as for the second one, this is the list of those files :-
administrator@DESKTOP-6HQEO4S:/mnt/d/veracruz/sdk/rust-examples$ git grep fd_create
fd-create/Cargo.toml:name = "fd_create"
fd-create/Cargo.toml:description = "Test of fd_create function"
fd-create/src/main.rs://! Test of fd_create function.
fd-create/src/main.rs:    let mut file = libveracruz::fd_create()?;
fd-create/test.sh:RUNARGS="--program target/wasm32-wasi/debug/fd_create.wasm --input-source target/wasm32-wasi/debug"

Before moving forward, I just wanted to confirm, the task is to open these files and change very mention of "fd_create" to "fd-create". 
Is this correct??

dominic-mulligan-arm commented:
Hi @AryanGodara, you need to first `cd ../../` into the Veracruz root directory before doing the `git grep`, otherwise you're only searching under `./sdk/rust_examples`.  I don't think any of those files are actually causing the issue that we are seeing with CI (and "fd_create" for the name of the test in the `Cargo.toml` file is actually idiomatic Rust, confusingly...).  You can also look at the CI build logs to understand what the issue is, at the top of this page...

AryanGodara commented:
@dominic-mulligan-arm Sorry for taking so long, but I've updated all instances of fd_create to fd-create
(Except the ones that were either constt declarations, for examples : 
const FD_CREATE_RUST_WASM: &'static str = "fd-create.wasm";   inside veracruz-server-test/src/main.rs

And I've commit the changes

ShaleXIONG commented on sdk/rust/libveracruz/src/lib.rs:
```diff
@@ -14,26 +14,26 @@ use std::os::wasi::io::{FromRawFd, RawFd};
     #[link(wasm_import_module = "veracruz_si")]
     extern "C" {
-        pub fn fd_create(x: u32) -> u32;
+        pub fn fd-create(x: u32) -> u32;
```
I doubt this will compile because of syntax error. You may have to revert some change back to `fd_create`. Note that there are difference between file directory and project names and programing syntax.

AryanGodara replied:
So, I need to ONLY change fd_create to fd-create when it is a 'directory name'??

ShaleXIONG replied:
I think so.

AryanGodara commented:
I've corrected my previous mistake, and reverted all the times when I changed a function name from fd_create to fd-create.
Now all the changes are in .toml or .cargo files, or in the comments mentioning the 'directory : rust-examples/fd-create'
One possible error now may by that fd_create.wasm has been changed to fd-create.wasm everywhere, since I noticed it is created according to the project directory's name.

If there are more errors in the tests, I'll try to rectify those mistakes also

dominic-mulligan-arm commented:
+1+1 = +1 reached, merging.
